### PR TITLE
Add pagination feature

### PR DESCRIPTION
**Pull request by cescoferraro**
This PR adds pagination feature

## Changes

- Returns x-total-count header to paginated responses
- uses Pagination component from antd
- refactor usePagination hook

## Fixes

- #1354 

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

**Review by mathnogueira — CHANGES_REQUESTED** [review]
I'm not seeing the changes on the `api/` folder. I think you forgot to add it to the PR.

**Review by jorgeepc — CHANGES_REQUESTED** [review]
Hey @cescoferraro I'm seeing an error when trying to search.
Screen Shot 2022-10-12 at 15 02 55

**Review by mathnogueira — APPROVED** [review]
Backend looks good

**Comment by cescoferraro** [comment]
@jorgeepc the new commit from @schoren fixed the issue you described

File: web/src/components/PaginatedList/PaginatedList.tsx [hunk 1]
```diff
@@ -13,35 +12,22 @@ type TParams<P> = P & {
...
-      hasNext={hasNext}
-      hasPrev={hasPrev}
-      isEmpty={isEmpty}
-      isFetching={isFetching}
-      isLoading={isLoading}
-      loadingComponent={<Loading />}
-      loadNext={loadNext}
-      loadPrev={loadPrev}
-    >
+    <Pagination emptyComponent={<Empty />} loadingComponent={<Loading />} {...pagination}>
```
Comment (jorgeepc) [inline web/src/components/PaginatedList/PaginatedList.tsx:28]: I see you added a `generic` for the `Pagination` component. but you are not passing it here.

**Review by jorgeepc — APPROVED** [review]
Looks good. Added a comment.

**Pull request merged**